chall01/osalmine.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/python3
+
+import sys
+
+morse = { 'A':'.-', 'B':'-...',
+                    'C':'-.-.', 'D':'-..', 'E':'.',
+                    'F':'..-.', 'G':'--.', 'H':'....',
+                    'I':'..', 'J':'.---', 'K':'-.-',
+                    'L':'.-..', 'M':'--', 'N':'-.',
+                    'O':'---', 'P':'.--.', 'Q':'--.-',
+                    'R':'.-.', 'S':'...', 'T':'-',
+                    'U':'..-', 'V':'...-', 'W':'.--',
+                    'X':'-..-', 'Y':'-.--', 'Z':'--..'}
+
+if len(sys.argv) != 2:
+	print("usage: ./osalmine.py <a-zA-Z string>")
+	quit()
+
+string = sys.argv[1].upper();
+if len(string) == 0:
+	print("usage: ./osalmine.py <a-zA-Z string>")
+	quit()
+
+if not all(x.isalpha() or x.isspace() for x in string):
+	print("usage: ./osalmine.py <a-zA-Z string>")
+	quit()
```

Comment:
All the checks for errors could be in one if or in one function. This way it's a little hard to read and also it breaks the main rule of programming: DRY (Don't Repeat Yourself)

Also, exactly in this check, you use the function `isspace()` which recognizes all kinds of spaces but later on you only check for one space sign. If someone will use '\t', your script will crash.